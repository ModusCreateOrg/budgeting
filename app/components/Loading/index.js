// @flow
import * as React from 'react';
import s from './styles.scss';

const Loading = () => (
  <div className={s.timelineItem}>
    <ul className={s.animatedBackground}>
      <li className={s.headerTop} />
      <li className={s.headerRight} />
      <li className={s.headerBottom} />
      <li className={s.subheaderRight} />
      <li className={s.subheaderBottom} />
      <li className={s.contentTop} />
      <li className={s.contentFirstEnd} />
      <li className={s.contentSecondLine} />
      <li className={s.contentSecondEnd} />
      <li className={s.contentThirdLine} />
      <li className={s.contentThirdEnd} />
    </ul>
  </div>
);

export default Loading;
