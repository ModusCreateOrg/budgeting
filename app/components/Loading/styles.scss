@-webkit-keyframes placeHolderShimmer {
  0% {
    background-position: -468px 0
  }
  100% {
    background-position: 468px 0
  }
}

@keyframes placeHolderShimmer {
  0% {
    background-position: -468px 0
  }
  100% {
    background-position: 468px 0
  }
}

.timelineItem {
  background: #fff;
  border: 1px solid;
  border-color: #e5e6e9 #dfe0e4 #d0d1d5;
  border-radius: 3px;
  padding: 12px;
  margin: 0 auto;
  min-height: 200px;
  min-width: 300px;
  max-width: 472px;
  width: 90vw;
  box-shadow: 0 1px 4px rgba(0, 0, 0, .16);
}

.animatedBackground {
  animation-duration: 1s;
  animation-fill-mode: forwards;
  animation-iteration-count: infinite;
  animation-name: placeHolderShimmer;
  animation-timing-function: linear;
  background: #f6f7f8;
  background: #eeeeee;
  background: linear-gradient(to right, #eeeeee 8%, #dddddd 18%, #eeeeee 33%);
  background-size: 800px 104px;
  height: 96px;
  position: relative;
  list-style: none;

  li {
    background: #fff;
    position: absolute;
    box-sizing: border-box;
  }
}

.headerTop,
.headerBottom,
.subheaderBottom {
  top: 0;
  left: 0;
  right: 0;
  height: 10px;
}

.headerRight,
.subheaderRight {
  top: 10px;
  left: 40px;
  height: 8px;
  width: 10px;
}

.headerBottom {
  top: 18px;
  height: 6px;
}

.subheaderRight {
  top: 24px;
  height: 6px;
}

.headerRight,
.subheaderRight {
  width: auto;
  left: 300px;
  right: 0;
}

.subheaderRight {
  left: 230px;
}

.subheaderBottom {
  top: 30px;
  height: 10px;
}

.contentTop,
.contentSecondLine,
.contentThirdLine,
.contentSecondEnd,
.contentThirdEnd,
.contentDirstEnd {
  top: 40px;
  left: 0;
  right: 0;
  height: 6px;
}

.contentTop {
  height: 20px;
}

.contentFirstEnd,
.contentSecondEnd,
.contentThirdEnd {
  width: auto;
  left: 380px;
  right: 0;
  top: 60px;
  height: 8px;
}

.contentSecondLine {
  top: 68px;
}

.contentSecondEnd {
  left: 420px;
  top: 74px;
}

.contentThirdLine {
  top: 82px;
}

.contentThirdEnd {
  left: 300px;
  top: 88px;
}