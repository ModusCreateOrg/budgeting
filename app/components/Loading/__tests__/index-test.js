import React from 'react';
import renderer from 'react-test-renderer';
import Loading from '../';

it('renders correctly', () => {
  const tree = renderer.create(<Loading />).toJSON();
  expect(tree).toMatchSnapshot();
});
