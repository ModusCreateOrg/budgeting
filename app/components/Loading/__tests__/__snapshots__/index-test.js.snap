// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="timelineItem"
>
  <ul
    className="animatedBackground"
  >
    <li
      className="headerTop"
    />
    <li
      className="headerRight"
    />
    <li
      className="headerBottom"
    />
    <li
      className="subheaderRight"
    />
    <li
      className="subheaderBottom"
    />
    <li
      className="contentTop"
    />
    <li
      className="contentFirstEnd"
    />
    <li
      className="contentSecondLine"
    />
    <li
      className="contentSecondEnd"
    />
    <li
      className="contentThirdLine"
    />
    <li
      className="contentThirdEnd"
    />
  </ul>
</div>
`;
